sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/postgres/PostgresInt2Tests.java

```diff
@@ -0,0 +1,397 @@
+                        "(5 rows)\n" +
+                        "\n"
+        );
+    }
+
+    @Test
+    public void testBadSelect() {
+        this.shouldFail("SELECT * FROM INT2_TBL AS s (a, b)", "List of column aliases must have same degree as table; table has 1 columns ('F1'), whereas alias list has 2 columns");
+
+    }
+
+    // We round differently than Postgres
+    // R: ("-1.5", -2)x1  L: ("-1.5", -1)x1
+    // R: ("-0.5", 0)x1   L: ("-0.5", 0)x1
+    // R: ("1.5", 2)x1    L: ("1.5", 1)x1
+    @Test @Ignore("fails because we round differently than Postgres")
+    public void testFloatRound() {
+        this.q(
+                "SELECT x, x::int2 AS int2_value " +
+                        "FROM (VALUES (-2.5::float8)," +
+                        "             (-1.5::float8)," +
+                        "             (-0.5::float8)," +
+                        "             (0.0::float8)," +
+                        "             (0.5::float8)," +
+                        "             (1.5::float8)," +
+                        "             (2.5::float8)) t(x);\n" +
+                        "  x   | int2_value \n" +
+                        "------+------------\n" +
+                        " -2.5 |         -2\n" +
+                        " -1.5 |         -2\n" +
+                        " -0.5 |          0\n" +
+                        "    0 |          0\n" +
+                        "  0.5 |          0\n" +
+                        "  1.5 |          2\n" +
+                        "  2.5 |          2"
+        );
+    }
+
+    // Fails because we round differently than Postgres
+    // R: (-2.5, -3)x1  L: (-2.5, -2)x1
+    // R: (-1.5, -2)x1  L: (-1.5, -1)x1
+    // R: (-0.5, -1)x1  L: (-0.5, 0)x1
+    // R: (0.5, 1)x1    L: (0.5, 0)x1
+    // R: (1.5, 2)x1    L: (1.5, 1)x1
+    // R: (2.5, 3)x1    L: (2.5, 2)x1
+    @Test @Ignore("fails because we round differently than Postgres")
+    public void testNumericRound() {
+        this.q(
+                "SELECT x, x::int2 AS int2_value " +
+                        "FROM (VALUES (-2.5::numeric)," +
+                        "             (-1.5::numeric)," +
+                        "             (-0.5::numeric)," +
+                        "             (0.0::numeric)," +
+                        "             (0.5::numeric)," +
+                        "             (1.5::numeric)," +
+                        "             (2.5::numeric)) t(x);\n" +
+                        "  x   | int2_value \n" +
+                        "------+------------\n" +
+                        " -2.5 |         -3\n" +
+                        " -1.5 |         -2\n" +
+                        " -0.5 |         -1\n" +
+                        "  0.0 |          0\n" +
+                        "  0.5 |          1\n" +
+                        "  1.5 |          2\n" +
+                        "  2.5 |          3"
+        );
+    }
+
+    @Test @Ignore("non decimals not suppported yet")
```

Comment:
This is not really about integers, but about literals.
Calcite does not support binary or hex literals.
I don't think we plan to support them either at this point.
I would suggest not creating tests for features that we don't plan to support.